package.json

```diff
@@ -8,6 +8,7 @@
         "test": "echo \"Error: no test specified\" && exit 1",
         "prepublish": "npm run build",
         "build": "npm run typecheck && babel --extensions=\".ts\" --out-dir=. src/",
+        "tsbuild": "tsc && babel --out-dir=. build/",
```

Comment:
Just wondering, but, why are you running both babel and tsc here?  
Also, what were your motivations behind adding this script?  

The reason I used babel over tsc for compilation was because it's much faster than the tsc compiler - understandably so, because babel doesn't do any type checking.  
However, I personally have a TS checker running in my IDEA eitherway, so I don't need typechecking in the development process.  

Nevertheless, let's settle on one of the two compilation methods for the build, because I don't really see the point of having a separate tsbuild command.

Reply:
My reasons are mostly the same as those given by [this person](https://github.com/AdamLeBlanc/gatsby-plugin-ts-loader#motivation). I'm used to tsc. I don't really trust @babel/plugin-transform-typescript, and knowing that it has some known issues/caveats, I'd prefer not to have the mental baggage of remembering precisely what they are. Plus, @babel/plugin-transform-typescript feels somehow unclean to me, like a hack. Tsc is the way nature intended.

The reason I do babel after tsc is to ensure that es6/es2015/es2017/esnext stuff is compatible. Probably not necessary for this plugin because there's no clientside code.

Is babel really faster when you are using typescript to typecheck beforehand anyway?

Reply:
The only issue that it has is that it doesn't typecheck.  
But when you're typechecking from your IDEA anyway, that's not really a problem, and I'll gladly take the (rather huge) speed boost it gives over TSC instead.

>Is babel really faster when you are using typescript to typecheck beforehand anyway?

Nope, the way it's set up now probably makes everything even slower because it's running both TSC & babel. Didn't put that much thought into it.  
I'm all in for switching the actual build to just using TSC only instead, but I prefer to keep babel for dev.

Reply:
Babel for dev and TSC for build sounds pretty reasonable. I might rebase my changes and put that as a separate pull request though, because I'm realising that including it in this PR is a bit of a trojan horse move.